### PR TITLE
Suggestions on modifying the crypto API

### PR DESCRIPTION
**Pull request by kevaundray**
With the new crypto API in 3038, there is now a clear delineation between the cryptography code and the client code. 

However, the client code will also need to implement interface methods on types defined in cryptography. This PR suggests a way to do this without modifying the existing behaviour of the code.

### CODE REVIEW

File: crypto/agg_kzg/agg_kzg.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+func (p BLSFieldElement) String() string {
+	return "0x" + hex.EncodeToString(p[:])
+}
+
+func (p *BLSFieldElement) UnmarshalText(text []byte) error {
+	return hexutil.UnmarshalFixedText("BLSFieldElement", text, p[:])
+}
+
+// Blob data
+type Blob [params.FieldElementsPerBlob]BLSFieldElement
```
Comment (roberto-bayardo) [inline crypto/agg_kzg/agg_kzg.go:43]: @Inphi mentioned we might change the base Blob class to be []bls.Fr underneath; I may look into this today.

Comment (roberto-bayardo) [reply crypto/agg_kzg/agg_kzg.go:43]: or rather he suggested making the geth Blob and the kzg_agg blob the same like you've done here. I guess the exact representation is TBD.  I've been using []Fr in my refactor on the KZG side.

Comment (kevaundray) [reply crypto/agg_kzg/agg_kzg.go:43]: I think we want to do a flat stream of bytes like you did here

Comment (shamatar) [reply crypto/agg_kzg/agg_kzg.go:43]: Ideally you want to generate another curve to fit 32 bytes injectively per scalar element

**Comment by roberto-bayardo** [comment]
looks like this one is now obsolete -- pls reopen if i'm mistaken.

**Pull request closed without merging**